### PR TITLE
Refactor price protection and other fixes

### LINKED ISSUE #2475: Bug: Max leverage issue when reducing position
### Description

When reducing position the modal will display an incorrect warning of max leverage exceeded on markets with leverage larger than 50x.

### Solution

Use the contract max leverage instead of app max leverage when reducing a position.

### PR DESCRIPTION
**Pull request by avclarke**
## Description

- Remove the extra desired fill price calculation when accepting large slippage
- Fix issue with max leverage when decreasing position (closes #2475)
- Adds some functionality to cancel expired delayed order when opening a new one (prevents separate manual cancel each time)

### CODE REVIEW

File: state/futures/actions.ts [hunk 1]
```diff
@@ -846,6 +845,25 @@ export const editCrossMarginPositionMargin = (
...
+	const marginDelta = selectCrossMarginMarginDelta(getState());
+	const { nativeSizeDelta } = selectCrossMarginTradeInputs(getState());
+
+	if (!marketInfo) throw new Error('No market selected');
+
+	dispatch(
+		stageCrossMarginTradePreview({
+			market: { key: marketInfo.marketKey, address: marketInfo.market },
+			orderPrice,
+			marginDelta: marginDelta,
```
Comment (LeifuChen) [inline state/futures/actions.ts:860]: A small detail - similar to `orderPrice`, we might just write `marginDelta` .

**Review by LeifuChen — COMMENTED** [review]
Looks good to me. Just found some eslint warnings.

File: components/TVChart/TVChart.tsx [hunk 2]
```diff
@@ -15,14 +15,14 @@ import { useAppSelector } from 'state/hooks';
 import { selectCurrentTheme } from 'state/preferences/selectors';
 import darkTheme from 'styles/theme/colors/dark';
 
+import { DEFAULT_RESOLUTION } from './constants';
```
[the remark this answers is not in the document]
Comment (avclarke) [reply components/TVChart/TVChart.tsx:18]: I think there's something up with my eslint version, it was telling me to move it. Will put it back though.

File: sections/futures/EditPositionModal/EditPositionSizeModal.tsx [hunk 3]
```diff
@@ -69,7 +69,10 @@ export default function EditPositionSizeModal() {
 		isFetchingPreview,
 	]);
 
-	const maxLeverage = useMemo(() => market?.appMaxLeverage ?? wei(1), [market?.appMaxLeverage]);
+	const maxLeverage = useMemo(
+		() => (editType === 0 ? market?.appMaxLeverage : market?.contractMaxLeverage) ?? wei(1),
+		[market?.appMaxLeverage, editType]
```
[the remark this answers is not in the document]
Comment (avclarke) [reply sections/futures/EditPositionModal/EditPositionSizeModal.tsx:74]: good catch, my eslint has been messed up, got lost in a bunch of other warnings.

**Review by platschi — APPROVED** [review]

**Pull request merged**